packages/ui/src/components/controls/Slider.tsx

```diff
@@ -0,0 +1,63 @@
+import {useEffect, useState} from 'preact/hooks';
+import {MouseButton, clamp} from '../../utils';
+import styles from './Controls.module.scss';
+
+export interface SliderProps {
+  value?: number;
+  onChange?: (value: number) => void;
+}
+
+export function Slider({value, onChange}: SliderProps) {
+  const [internalValue, setInternalValue] = useState(value);
+
+  useEffect(() => {
+    setInternalValue(value);
+  }, [value]);
+
+  return (
+    <div
+      className={styles.slider}
+      onPointerDown={event => {
+        if (event.button === MouseButton.Left) {
+          event.preventDefault();
+          event.stopPropagation();
+          event.currentTarget.setPointerCapture(event.pointerId);
+
+          const rect = event.currentTarget.getBoundingClientRect();
+          // +2 & +4 to account for thumb offset
+          const y = event.clientY - rect.y + 2;
+          const newInternalValue = 1 - y / (rect.height + 4);
+          setInternalValue(clamp(0, 1, newInternalValue));
```

Comment:
The thumb offset is not necessary because the thumb should be able to overflow the slider in extreme positions:
![image](https://github.com/motion-canvas/motion-canvas/assets/64662184/d64e4798-88b1-4f76-be76-30b341fe3dea)

Reply:
The idea is that the thumb itself is clickable as well. So for a track of length `120px`, the clickable range would actually be `-4px` - `124px`, relative to the track (for the case of a `8px` wide circle).

That's why I add the offset and then divide by `height + circle width`.

But looking at it now, clamping `y` to stay between `0` and `height` might be a better solution.